ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/BlsToExecutionOperationPoolTest.java

```diff
@@ -0,0 +1,265 @@
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.TestSpecFactory;
+import tech.pegasys.teku.spec.datastructures.blocks.blockbody.BeaconBlockBodySchema;
+import tech.pegasys.teku.spec.datastructures.blocks.blockbody.versions.capella.BeaconBlockBodySchemaCapella;
+import tech.pegasys.teku.spec.datastructures.operations.SignedBlsToExecutionChange;
+import tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState;
+import tech.pegasys.teku.spec.logic.common.operations.validation.OperationInvalidReason;
+import tech.pegasys.teku.spec.util.DataStructureUtil;
+import tech.pegasys.teku.statetransition.validation.InternalValidationResult;
+import tech.pegasys.teku.statetransition.validation.SignedBlsToExecutionChangeValidator;
+
+public class BlsToExecutionOperationPoolTest {
+
+  private final Spec spec = TestSpecFactory.createMinimalCapella();
+  private final DataStructureUtil dataStructureUtil = new DataStructureUtil(spec);
+  private final Function<UInt64, BeaconBlockBodySchema<?>> beaconBlockSchemaSupplier =
+      slot -> spec.atSlot(slot).getSchemaDefinitions().getBeaconBlockBodySchema();
+  private final BeaconState state = mock(BeaconState.class);
+  private final StubMetricsSystem metricsSystem = new StubMetricsSystem();
+  private final SignedBlsToExecutionChangeValidator validator =
+      mock(SignedBlsToExecutionChangeValidator.class);
+  private final Function<UInt64, SszListSchema<SignedBlsToExecutionChange, ?>>
+      blsToExecutionSchemaSupplier =
+          beaconBlockSchemaSupplier
+              .andThen(BeaconBlockBodySchema::toVersionCapella)
+              .andThen(Optional::orElseThrow)
+              .andThen(BeaconBlockBodySchemaCapella::getBlsToExecutionChangesSchema);
+  private final OperationPool<SignedBlsToExecutionChange> pool =
+      new BlsToExecutionOperationPool(
+          "BlsToExecutionOperationPool", metricsSystem, blsToExecutionSchemaSupplier, validator);
+
+  @BeforeEach
+  void init() {
+    when(state.getSlot()).thenReturn(UInt64.ZERO);
+  }
+
+  @Test
+  void emptyPoolShouldReturnEmptyList() {
+    assertThat(pool.getItemsForBlock(state)).isEmpty();
+  }
+
+  @Test
+  void shouldAddEntryToPool() {
+    when(validator.validateForGossip(any()))
+        .thenReturn(SafeFuture.completedFuture(InternalValidationResult.ACCEPT));
+    final SignedBlsToExecutionChange item = dataStructureUtil.randomSignedBlsToExecutionChange();
+    final SafeFuture<?> future = pool.addLocal(item);
+    assertThat(future.isCompletedNormally()).isTrue();
+    assertThat(pool.size()).isEqualTo(1);
+    assertThat(pool.getAll()).containsExactly(item);
+  }
+
+  @Test
+  void shouldRejectDuplicateEntryFromPool() throws ExecutionException, InterruptedException {
+    final SignedBlsToExecutionChange item = initPoolWithSingleItem();
+
+    final SafeFuture<InternalValidationResult> future2 = pool.addLocal(item);
+    assertThat(future2.get().code()).isEqualTo(InternalValidationResult.IGNORE.code());
```

Comment:
NIT

Could use:
```
assertThat(future2).isCompletedWithValueMatching(InternalValidationResult::isIgnore);
```

Or, if we want to be certain we are ignoring the message based on the expected reason, we could test if the ignore description matches what we would expect:

```
  final InternalValidationResult expectedValidationResult =
      InternalValidationResult.ignore("BlsToExecutionChange is not the first one for validator %s.",
      item.getMessage().getValidatorIndex());
  
  assertThat(future2).isCompletedWithValue(expectedValidationResult);
```